test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJmap.java

```diff
@@ -0,0 +1,183 @@
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import com.ibm.tools.attach.attacher.OpenJ9VirtualMachine;
+import com.sun.tools.attach.AttachNotSupportedException;
+import com.sun.tools.attach.VirtualMachine;
+
+/**
+ * Test the jmap utility.
+ */
+@Test(groups = { "level.extended" })
+public class TestJmap extends AttachApiTest {
+
+	private static final String JMAP_COMMAND = "jmap"; //$NON-NLS-1$
+	private static String myId;
+	
+	/* test objects */
+	private ArrayList<HelloWorld> hws;
+	private int[][] primitiveArray;
+	private long[] primitiveVector;
+	private String[][] objectArray;
+	private Integer[] objectVector;
+
+	@Test
+	public void testHisto() throws IOException {
+		List<String> jmapOutput = runCommand(Arrays.asList(myId, "-histo")); //$NON-NLS-1$
+		logJmapOutput(jmapOutput);
+		checkHeapStats(jmapOutput);
+	}
+
+	@Test
+	public void testHistoLive() throws IOException {
+		List<String> jmapOutput = runCommand(Arrays.asList(myId, "-histo:live")); //$NON-NLS-1$
+		logJmapOutput(jmapOutput);
+		checkHeapStats(jmapOutput);
+	}
+
+	@Test
+	public void testHeapHisto() throws AttachNotSupportedException, IOException {
+		TargetManager tgtMgr = new TargetManager(TARGET_VM_CLASS, null);
+		assertTrue(AttachApiTest.CHILD_PROCESS_DID_NOT_LAUNCH, tgtMgr.syncWithTarget());
+		VirtualMachine vm = VirtualMachine.attach(tgtMgr.targetId);
+		assertTrue(vm instanceof OpenJ9VirtualMachine, "Wrong attach API VirtualMachine"); //$NON-NLS-1$
+		OpenJ9VirtualMachine myVm = (OpenJ9VirtualMachine) vm;
+		InputStream result = myVm.heapHisto();
+		checkForProperties(result);
+		result = myVm.heapHisto("-live"); //$NON-NLS-1$
+		checkForProperties(result);
+		result = myVm.heapHisto("-all"); //$NON-NLS-1$
+		checkForProperties(result);
+		vm.detach();
+		tgtMgr.terminateTarget();
+	}
+
+	@BeforeSuite
+	protected void setupSuite() {
+		assertTrue("This test is valid only on OpenJ9", //$NON-NLS-1$
+				System.getProperty("java.vm.vendor").contains("OpenJ9")); //$NON-NLS-1$//$NON-NLS-2$
+		getJdkUtilityPath(JMAP_COMMAND);
+		myId = TargetManager.getVmId();
+		assertNotNull(myId, "Attach ID missing"); //$NON-NLS-1$
+		log(myId);
+		
+		/* make test objects */
+		hws = new ArrayList<>();
+		for (int i = 0; i < 3; i++) {
+			hws.add(new HelloWorld());
+		}
+		objectArray = new String[][] {{"a", "b"}, {"c", "d"}}; //$NON-NLS-1$//$NON-NLS-2$ //$NON-NLS-3$//$NON-NLS-4$
+		objectVector = new Integer[] {Integer.valueOf(10), Integer.valueOf(20)};
+		primitiveArray = new int[][] {{1, 2}, {3, 4}};
+		primitiveVector = new long[] {5, 6};
+	}
+
+	private void checkHeapStats(List<String> heapStats) {
+		String hwClassName = HelloWorld.class.getName();
+		String expectedClasses[] = new String[] {hwClassName, 
+				primitiveArray.getClass().getName(),
+				primitiveVector.getClass().getName(),
+				objectArray.getClass().getName(),
+				objectVector.getClass().getName(),
+		};
+		for (String className: expectedClasses) {
```

Comment:
formattting: space around `:`

Reply:
Thanks.  I also moved the initialization of `expectedClasses` (and fixed the declaration) to `setupSuite()` so it runs only once.